pkg/apis/batch/validation/validation.go

```diff
@@ -99,12 +105,18 @@ func ValidateGeneratedSelector(obj *batch.Job) field.ErrorList {
 	// have placed certain labels on the pod, but this could have failed if
 	// the user added coflicting labels.  Validate that the expected
 	// generated ones are there.
+	allErrs = append(allErrs, apivalidation.ValidateHasLabel(obj.Spec.Template.ObjectMeta, field.NewPath("spec").Child("template").Child("metadata"), controllerUid, string(obj.UID))...)
```

Comment:
This change is adding extra validation, which is generally hard.

Note that this function is called during JobUpdates. Then every Job update would fail for old jobs.

In general, we need to allow this validation to pass if it was passing before the update. See this guide for a similar change when adding enums https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#new-enum-value-in-existing-field

Reply:
I read the guide and I am still a bit confused on this.

So for this release, should I just remove the validation with the new labels? Until the labels are present in a previous release I can't really validate them?